middle_end/flambda2/simplify_shared/inlining_report.ml

```diff
@@ -564,67 +581,93 @@ module Inlining_tree = struct
     Format.fprintf ppf "@[<h>%a@ %s@ %a%a%a@]@,@," stars depth label f w
       print_dbg dbg print_uid uid
 
-  let print_decisions ?callee ~compilation_unit ppf decisions =
+  let print_decision_with_context ppf decision_with_context =
+    Format.fprintf ppf "@[%a@]@," Decision_with_context.print
+      decision_with_context
+
+  let print_decisions ppf decisions =
     match decisions with
-    | [] -> begin
-      match callee with
-      | None -> Format.fprintf ppf "Unknown"
-      | Some def ->
-        let defined_in = Inlining_history.Absolute.compilation_unit def in
-        if Compilation_unit.equal defined_in compilation_unit
-        then Format.fprintf ppf "Unknown"
-        else
-          Format.fprintf ppf
-            "@[<hov>This@ decision@ was@ taken@ while@ compiling@ %s.@]"
-            (Compilation_unit.string_for_printing defined_in)
-    end
+    | [] -> assert false
     | decisions ->
       let decisions = List.rev decisions in
-      Format.fprintf ppf "@[<v>@]";
-      List.iter
-        (fun decision_with_context ->
-          Format.fprintf ppf "@[%a@]@," Decision_with_context.print
-            decision_with_context)
-        decisions;
+      Format.fprintf ppf "@[<v>";
+      List.iter (print_decision_with_context ppf) decisions;
       Format.fprintf ppf "@]"
 
-  let rec print ~compilation_unit ~depth ppf t =
+  let rebuild_path ~path ~dbg (key : Key.element) =
+    match key with
+    | Scope (Unknown, _) -> IHA.Unknown { prev = path }
+    | Scope (Module, name) -> IHA.Module { name; prev = path }
+    | Scope (Class, name) -> IHA.Class { name; prev = path }
+    | Call callee -> IHA.Call { callee; dbg; prev = path }
+    | Fundecl fundecl -> IHA.Function { name = fundecl; dbg; prev = path }
+
+  let print_reference ~compilation_unit ppf to_ =
+    let cu ppf () =
+      let defined_in = IHA.compilation_unit to_ in
+      if Compilation_unit.equal defined_in compilation_unit
+      then Format.fprintf ppf "this compilation unit"
+      else
+        Format.fprintf ppf
+          "@[<hov>This@ decision@ was@ taken@ while@ compiling@ %s.@]"
+          (Compilation_unit.string_for_printing defined_in)
+    in
+    Format.fprintf ppf
+      "@[<v>The decision to inline this call was taken in %a at %a.@]" cu ()
```

Comment:
This probably needs the same wrapping as in `cu`? With the `<hov>` and `@`s?

Reply:
Fixed